src/ape_accounts/accounts.py

```diff
@@ -119,17 +120,20 @@ def delete(self):
         self.__decrypt_keyfile(passphrase)
         self.keyfile_path.unlink()
 
-    def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
-        user_approves = self.__autosign or click.confirm(f"{msg}\n\nSign: ")
-        if not user_approves:
-            return None
-
-        signed_msg = EthAccount.sign_message(msg, self.__key)
-        return MessageSignature(
-            v=signed_msg.v,
-            r=to_bytes(signed_msg.r),
-            s=to_bytes(signed_msg.s),
-        )
+    def sign_message(self, msg: Any, **signer_options) -> Optional[MessageSignature]:
+        if isinstance(msg, str):
+            msg = encode_defunct(text=msg)
+        if isinstance(msg, SignableMessage):
+            user_approves = self.__autosign or click.confirm(f"{msg}\n\nSign: ")
+            if not user_approves:
+                return None
+            signed_msg = EthAccount.sign_message(msg, self.__key)
+            return MessageSignature(
+                v=signed_msg.v,
+                r=to_bytes(signed_msg.r),
+                s=to_bytes(signed_msg.s),
+            )
+        raise TypeError(f"Can't sign message of type {type(msg)}")
```

Comment:
Can return `None` to signify it didn't sign as well

Reply:
Wonder which is better? Remove `Optional` from the signature, and always raise if it can't sign? Or return None

Reply:
Is there value in distinguishing between type mismatch and other signature failures (like user rejecting to sign)? We can distinguish between the two by raishing instead of returning None, but maybe we don't need/want to do that

Reply:
To help debug the issue, it would be helpful to see that feedback. But None might be more useful for multi-signing flows

Perhaps it can be a warning logg instead?

Reply:
Makes sense, just made that change